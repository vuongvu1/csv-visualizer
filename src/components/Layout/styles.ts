import styled from "styled-components";
import { neutralColor } from "configs/colors";
import { ScreenWidth } from "configs/styles";

const HEADER_HEIGHT = "100px";

export default {
  Container: styled.div`
    width: 100%;
    min-height: 100vh;
    display: flex;
    flex-direction: column;
  `,
  Header: styled.div`
    width: 100%;
    height: ${HEADER_HEIGHT};
    display: flex;
    justify-content: center;
    align-items: center;
  `,
  BodyContainer: styled.div`
    flex: 1;
    width: 100%;
    background-color: ${neutralColor};
    display: flex;
    justify-content: center;
  `,
  Body: styled.div`
    width: 100%;
    max-width: 930px;
    display: flex;

    @media (max-width: ${ScreenWidth.medium}) {
      flex-direction: column;
    }
  `,
};
