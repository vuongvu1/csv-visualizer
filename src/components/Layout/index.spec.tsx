import React from "react";
import { render, cleanup } from "@testing-library/react";
import Layout from "./index";

describe("Layout", () => {
  afterEach(cleanup);

  test("should match snapshot", () => {
    const { container } = render(<Layout title="Page Title">Children</Layout>);
    expect(container).toMatchSnapshot();
  });
});
