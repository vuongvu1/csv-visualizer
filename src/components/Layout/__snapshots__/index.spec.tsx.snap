// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should match snapshot 1`] = `
.c0 {
  width: 100%;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  width: 100%;
  height: 70px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  width: 100%;
  background-color: #EFEFEF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c4 {
  width: 100%;
  max-width: 930px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  color: #EFEFEF;
  font-weight: normal;
  font-size: 16px;
  color: #3F7EF0;
  font-weight: bold;
  font-size: 32px;
}

@media (max-width:768px) {
  .c4 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <h1
        class="c2"
        type="h1"
      >
        Page Title
      </h1>
    </div>
    <div
      class="c3"
    >
      <div
        class="c4"
      >
        Children
      </div>
    </div>
  </div>
</div>
`;
